test/components/rankers/test_hugging_face_tei.py

```diff
@@ -0,0 +1,291 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+import pytest
+from unittest.mock import patch, MagicMock
+
+import requests
+import httpx
+
+from haystack import Document
+from haystack.components.rankers.hugging_face_tei import HuggingFaceTEIRanker, TruncationDirection
+from haystack.utils import Secret
+
+
```

Comment:
in most unit tests, you may need to set
`monkeypatch.delenv("HF_API_TOKEN", raising=False)`
`monkeypatch.delenv("HF_TOKEN", raising=False)`

See https://github.com/deepset-ai/haystack/issues/8811 for more details